core/domain/takeout_service_test.py

```diff
@@ -876,6 +929,180 @@ def test_export_data_for_full_user_trivial_is_correct(self):
         expected_images = []
         self.assertEqual(expected_images, observed_images)
 
+    def test_exports_have_single_takeout_dict(self):
+        """Test to ensure that all export policies that specify a key for the
+        Takeout dict are also models that specify this policy are type
+        MULTIPLE_INSTANCES_PER_USER.
+        """
+        self.set_up_non_trivial()
+
+        # We set up the feedback_thread_model here so that we can easily
+        # access it when computing the expected data later.
+        feedback_thread_model = feedback_models.GeneralFeedbackThreadModel(
+            entity_type=self.THREAD_ENTITY_TYPE,
+            entity_id=self.THREAD_ENTITY_ID,
+            original_author_id=self.USER_ID_1,
+            status=self.THREAD_STATUS,
+            subject=self.THREAD_SUBJECT,
+            has_suggestion=self.THREAD_HAS_SUGGESTION,
+            summary=self.THREAD_SUMMARY,
+            message_count=self.THREAD_MESSAGE_COUNT
+        )
+        feedback_thread_model.put()
+
+        thread_id = feedback_services.create_thread(
+            self.THREAD_ENTITY_TYPE,
+            self.THREAD_ENTITY_ID,
+            self.USER_ID_1,
+            self.THREAD_SUBJECT,
+            self.MESSAGE_TEXT
+        )
+        feedback_services.create_message(
+            thread_id,
+            self.USER_ID_1,
+            self.THREAD_STATUS,
+            self.THREAD_SUBJECT,
+            self.MESSAGE_TEXT
+        )
+
+        # Retrieve all models for export.
+        all_models = [
+            clazz
+            for clazz in self._get_model_classes()
+            if not clazz.__name__ in self.BASE_CLASSES
+        ]
+
+        for model in all_models:
+            export_method = model.get_model_association_to_user()
+            export_policy = model.get_export_policy()
+            num_takeout_keys = 0
+            for field_export_policy in export_policy.values():
+                if (field_export_policy ==
+                        base_models
+                        .EXPORT_POLICY
+                        .EXPORTED_AS_KEY_FOR_TAKEOUT_DICT):
+                    num_takeout_keys += 1
+
+            if (export_method ==
+                    base_models.MODEL_ASSOCIATION_TO_USER
+                    .MULTIPLE_INSTANCES_PER_USER):
+                self.assertLessEqual(num_takeout_keys, 1)
```

Comment:
Should this be assertEqual? 

(If not, maybe add some explanation about why there might be 0 of these, with specific examples.)

Reply:
An example is the GeneralFeedbackThreadModel, where we have the feedback_model.id as the key to export, but the feedback_model.id is not an explicit field in the GeneraFeedbackThreadModel (see the export_data function for more details)

Reply:
Oh, that's interesting. Here's what I suggest: add a property like ID_IS_USED_AS_TAKEOUT_KEY to all models. It is False in the base class. Specific classes, like GeneralFeedbackThreadModel, override it to True. Then, verify here that:
- if the boolean is True, the export has 0 keys
- otherwise, it has 1 key

Reply:
(This assumes that usage of ID is the only case where this happens.)

Reply:
Yes, that's true. Sounds good, I'll make the change.

Reply:
Sorry, to clarify. Where would I be adding this variable? And where would I be changing it?

Reply:
I don't see any examples of model specific constants (but I might just be missing them)

Reply:
Would it make more sense for this to be a method?

Reply:
It's really a boolean in spirit, so ... probably better for it to be a constant.

See `REQUIRE_PAYLOAD_CSRF_CHECK` -- does that help as an example of a class-level constant? That's the sort of thing I was thinking of.

Reply:
done